breizorro/config/breizorro.yaml

```diff
@@ -0,0 +1,120 @@
+opts:
+  log:
+    dir: logs/log-{config.run.datetime}
+    name: log-{info.taskname}
+    nest: 10
+    symlink: log
+
+cabs:
+  breizorro:
+    info: "Mask creation and manipulation for radio astronomy images (https://github.com/ratt-ru/breizorro)"
+    command: breizorro
+    inputs:
+      restored_image:
+        info: "Restored image file from which to build the mask"
+        dtype: File
+        required: false
+      mask_image:
+        info: "Input mask file(s). Either restored-image or mask-image must be specified."
+        dtype: File
+        required: false
+      threshold:
+        info: "Sigma threshold for masking (default = 6.5)"
+        dtype: float
+        required: false
+        default: 6.5
+      boxsize:
+        info: "Box size over which to compute stats (default = 50)"
+        dtype: int
+        required: false
+        default: 50
+      savenoise:
+        info: "Export noise image as FITS file"
+        dtype: bool
+        required: false
+        default: false
+      merge:
+        info: "Merge one or more masks or region files"
+        dtype: list[File]
```

Comment:
I think this should be ``List[File]``, also elsewhere. Also, for multiple options, do we want ``--merge FILE1 FILE2``, ``--merge FILE1,FILE2`` or ``--merge FILE1 --merge FILE2``? Set ``policies: repeat`` accordingly, to ``list``, ``[]`` or ``repeat``.

Reply:
Will the following work?
```
      merge:
        info: "Merge one or more masks or region files"
        dtype: List[File]
        policies:
          repeat: ' '
```

When repeat is set to `list` it passes the argument as a `string` instead of `list` of files.
Also, this results in no file validation.

Reply:
Actually I lie, ``--merge FILE1 FILE2`` is not available with click. So you have to decide on ``--merge 'FILE1 FILE2'`` (cumbersome because it needs shell quotes), ``--merge FILE1,FILE2``, or --merge ``FILE1 --merge FILE2``.

> When repeat is set to `list` it passes the argument as a `string` instead of `list` of files.
> Also, this results in no file validation.

File validation is not done via clickify anyway. But it's strange that it passes as a string. Could you please test again? Alternatively, try the ``clickify-cleanup`` branch I just pushed to stimela. I've added a test there and it seems to work as expected:

```
(stimela) oms@baker:~/projects/stimela2/tests/scabha_tests$ ./test_clickify.py 1 2 a b --files1 'x y z' --files2 x --files2 y --files3 a,b --output x
1 2 None 2.0 None
remainder: ('a', 'b')
files1: ['x', 'y', 'z']
files2: ('x', 'y')
files3: ['a', 'b']
output: x
(stimela) oms@baker:~/projects/stimela2/tests/scabha_tests$ 
```

Reply:
> Alternatively, try the clickify-cleanup branch I just pushed to stimela. I've added a test there and it seems to work as expected:

Now I get the following when I set repeat policy to `list`: `scabha.exceptions.SchemaError: click parameter 'merge': repeat=list policy is only supported for positional=true parameters`

I'd go with `--merge File1,File2`.

```
      merge:
        info: "Merge one or more masks or region files"
        dtype: List[File]
        metavar: "MASK(s)|REG(s)"
        policies:
          repeat: '[]'
```

But for the `extract-islands`/`remove-islands` parameters which were previously specified as `--remove-islands 19 21 14h16m20.3s,-65d42m07s`,  can now be configured as:

```
      remove_islands:
        info: "Remove islands from input mask (list by number or coordinates)"
        dtype: Optional[Tuple[int, str]]
        policies:
          repeat: ' '
```

which results in `--remove-islands '19 21 14h16m20.3s,-65d42m07s'`

Reply:
> 
> Now I get the following when I set repeat policy to `list`: `scabha.exceptions.SchemaError: click parameter 'merge': repeat=list policy is only supported for positional=true parameters`

Yeah that's as expected. Click does not support variable-length options -- only positional arguments (i.e. the last positional argument can be variable-length, it will swallow "the rest" of the command line.)

> But for the `extract-islands`/`remove-islands` parameters which were previously specified as `--remove-islands 19 21 14h16m20.3s,-65d42m07s`, can now be configured as:
> 
> ```
>       remove_islands:
>         info: "Remove islands from input mask (list by number or coordinates)"
>         dtype: Optional[Tuple[int, str]]
>         policies:
>           repeat: ' '
> ```
> 
> which results in `--remove-islands '19 21 14h16m20.3s,-65d42m07s'`

I'm confused that this works. ``Tuple[int, str]`` should expect two arguments, an int and a str, so if it accepts three, that's a bug in ``clickify_parameters``. I'll investigate. In any case you probably meant ``List[Union[int, str]]``, right?

I'm a bit concerned that we have two different ways of specifying multiple arguments (comma-separated and space-separated) in the CLI now, this is potentially confusing. Maybe consider using ``:`` as a separator throughout?

Reply:
I released stimela 2.0.2 which incorporates the CLI fixes.

Reply:
> I'm confused that this works. Tuple[int, str] should expect two arguments, an int and a str, so if it accepts three, that's a bug in clickify_parameters.

I was yet to test this dtype, actually I had it as `List[str]`. 

>  I'll investigate. In any case you probably meant List[Union[int, str]], right?

Exactly.

> I'm a bit concerned that we have two different ways of specifying multiple arguments (comma-separated and space-separated) in the CLI now, this is potentially confusing. Maybe consider using : as a separator throughout?

Sounds good to me.

Example:
`
breizorro -r mypipelinerun-MFS-image.fits--merge mask.fits,adass.reg --number-islands  --remove-islands 19,21,14h16m20.3:-65d42m07
`

```
breizorro.breizorro - 2024-12-20 00:55:47,393 INFO - Welcome to breizorro
breizorro.breizorro - 2024-12-20 00:55:47,444 INFO - Version: 0.1.3
breizorro.breizorro - 2024-12-20 00:55:47,444 INFO - Usage: breizorro --help
breizorro.breizorro - 2024-12-20 00:55:47,457 INFO - Generating mask using threshold 6.5
breizorro.breizorro - 2024-12-20 00:55:47,457 INFO - Generating noise map
breizorro.breizorro - 2024-12-20 00:55:47,582 INFO - Median noise value is 0.0007369568920694292
WARNING: FITSFixedWarning: 'datfix' made the change 'Set MJD-OBS to 58318.744319 from DATE-OBS'. [astropy.wcs.wcs]
breizorro.breizorro - 2024-12-20 00:55:47,615 INFO - Treating mask.fits as a FITS mask
breizorro.breizorro - 2024-12-20 00:55:47,622 INFO - Merged into mask
breizorro.breizorro - 2024-12-20 00:55:47,894 INFO - Merging in 24 regions from adass.reg
breizorro.breizorro - 2024-12-20 00:55:48,506 INFO - (Re)numbering islands
breizorro.breizorro - 2024-12-20 00:55:48,524 INFO - Number of islands: 35
breizorro.breizorro - 2024-12-20 00:55:48,524 INFO - Removing islands: ['19', '21', '14h16m20.3:-65d42m07']
breizorro.breizorro - 2024-12-20 00:55:48,550 INFO - coordinates <SkyCoord (ICRS): (ra, dec) in deg
    (214.08458333, -65.70194444)> correspond to pixel 328, 241 with value 5
breizorro.breizorro - 2024-12-20 00:55:48,568 INFO - Writing mypipelinerun-MFS-image.mask.fits
breizorro.breizorro - 2024-12-20 00:55:48,588 INFO - Done
```